### PR TITLE
Add a simple functional test for transactions relaying

### PR DESCRIPTION
**Pull request by stanislav-tkach**

### CODE REVIEW

File: test/functional/mempool_submit_tx.py [hunk 1]
```diff
@@ -11,8 +11,7 @@
 """
 
 from test_framework.test_framework import BitcoinTestFramework
-from test_framework.util import (assert_equal, assert_raises_rpc_error)
-from test_framework.authproxy import JSONRPCException
+from test_framework.util import (assert_raises_rpc_error)
```
Comment (TheQuantumPhysicist) [inline test/functional/mempool_submit_tx.py:14]: Parenthesis are not required

**Review by TheQuantumPhysicist — APPROVED** [review]

File: test/functional/p2p_relay_transactions.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            'inputs': [],
+            'outputs': [],
+            'time_lock': 0,
+        }
+        signed_tx = {
+            'transaction': tx,
+            'signatures': [],
+        }
+        encoded_tx = signed_tx_obj.encode(signed_tx).to_hex()[2:]
+        tx_id = scalecodec.ScaleBytes(mintlayer_hash(base_tx_obj.encode(tx).data)).to_hex()[2:]
```
Comment (iljakuklic) [inline test/functional/p2p_relay_transactions.py:46]: We should create some helper methods to encode stuff into hex and get hex ID of an object. This is a common task and is already duplicated in the mempool test.

**Review by iljakuklic — APPROVED** [review]
+1

**Review by muursh — APPROVED** [review]

File: test/functional/p2p_relay_transactions.py (hunk 2 above)
Comment (TheQuantumPhysicist) [reply test/functional/p2p_relay_transactions.py:46]: Agreed. Someone please open the issue and explain the problem in better details. I didn't do this often so I can't describe it properly.

Comment (stanislav-tkach) [reply test/functional/p2p_relay_transactions.py:46]: https://github.com/mintlayer/mintlayer-core/issues/849

**Comment by stanislav-tkach** [comment]
I have removed the `is_initial_block_download` RPC method in favor of the new `(chainstate_)info` method.

**Comment by TheQuantumPhysicist** [comment]
Ready to merge?

**Comment by stanislav-tkach** [comment]
I think so.

**Pull request merged**